### ISSUE TITLE
SendSmtpEmail.ScheduledAt has type `time.Time`

### ISSUE THREAD

**Issue by alfa-alex** [sendinblue/APIv3-go-library#32]
### Summary
The recent release `v2.1.0` introduced a field `ScheduledAt time.Time` on the `lib.SendSmtpEmail` struct. This breaks existing code.

### Background
Unfortunately, the Go JSON encoder is unable to omit `time.Time` zero values. Instead, zero values of `time.Time` are marshalled into `"0001-01-01T00:00:00Z"`.
For example,
```go
json.NewEncoder(os.Stdout).Encode(lib.SendSmtpEmail{})
```
will output
```
{"scheduledAt":"0001-01-01T00:00:00Z"}
```

### Problem
The latest release of `sendinblue/APIv3-go-library` introduced the following field:
```go
// UTC date-time on which the email has to schedule (YYYY-MM-DDTHH:mm:ss.SSSZ). Prefer to pass your timezone in date-time format for scheduling. There can be an expected delay of +5 minutes in scheduled email delivery. **Please note this feature is currently a public beta**.
ScheduledAt time.Time `json:"scheduledAt,omitempty"`
```
By solely updating the sendinblue Go SDK, previously working code to send emails will no longer work because
- this field will have the zero value by default
- will be marshalled to `0001-01-01T00:00:00Z`
- sendinblue will return a `400` error `Scheduled date should not be less than current date`

### Side note
Other structs do have a `ScheduledAt` field already (e.g. `CreateEmailCampaign` or `CreateSmsCampaign`) but there this field is generated as a `string`, not a `time.Time`. I guess the reason was to circumvent the above problem. Probably this is the way to go for the `SendSmtpEmail` as well.

**Comment by LandazuriPaul** [comment]
I had the exact same issue.

It is very surprising to see a feature which is supposed to be in **beta** from [SendInBlue's own documentation](https://developers.sendinblue.com/reference/sendtransacemail) (see screenshot below) breaking code which is in **production**.

<img width="604" alt="Capture d’écran 2022-08-01 à 19 48 57" src="https://user-images.githubusercontent.com/5787964/182221959-6b1d3916-c7f7-465d-9263-31d31d12ebff.png">

After the previous issue (#23) having to do with a mistyped API exposed by this library, it looks like SendInBlue doesn't test its own SDK at all. To be honest, this doesn't instil confidence in your product and the way you maintain it. As I'm currently working on a prototype, I was hoping to integrate your SDK but I'm now much more likely to go for a competitor which takes care of its SDK. This is a shame since I really wanted to use your service.

**Comment by jain-raunak** [comment]
Hello All,
Apologies for the inconveniences occurred.

This issue occurred because we generate our all SDKs through swagger codegen and omitempty will not work for this as its default zero value is 0001-01-01T00:00:00Z. We have created a task to resolve this issue and will update once it gets resolved meanwhile you can use **scheduledAt.IsZero()** to resolve the blocker.

**Comment by Codehardt** [comment]
same issue here .. had to remove scheduledAt from code manually

**Comment by bondhan** [comment]
Hi @jain-raunak when can we expect this to be fixed?

**Comment by jain-raunak** [comment]
Hello,
We will pick this task in the very next sprint (starting from 12th Dec). Thanks!

**Comment by jain-raunak** [comment]
Hello,
This issue got fixed in latest updated version v2.1.2, please verify the same.
Thanks!
